hazelcast/include/hazelcast/client/protocol/ClientMessage.h

```diff
@@ -795,6 +876,204 @@ class HAZELCAST_API ClientMessage
         return h;
     }
 
+    /**
+     * Reads the header of the current frame.
+     * The cursor must be at a frame's beginning.
+     */
+    frame_header_t read_frame_header()
+    {
+        frame_header_t header{};
+        auto pos = rd_ptr(SIZE_OF_FRAME_LENGTH_AND_FLAGS);
+        std::memcpy(&header.frame_len, pos, sizeof(header.frame_len));
+        pos += sizeof(header.frame_len);
+        std::memcpy(&header.flags, pos, sizeof(header.flags));
+        return header;
+    }
+
+    template<typename T>
+    typename std::enable_if<std::is_same<T, sql::sql_column_metadata>::value,
+                            T>::type
+    get()
+    {
+        // skip begin frame
+        skip_frame();
+
+        const frame_header_t header = read_frame_header();
+
+        auto type = get<int32_t>();
+
+        bool nullable = true;
+        int nullable_size = 0;
+        bool nullable_exist = false;
+        if (header.frame_len - SIZE_OF_FRAME_LENGTH_AND_FLAGS >=
+            INT32_SIZE + INT8_SIZE) {
```

Comment:
Would it make sense to introduce `BOOL_SIZE` instead of `INT8_SIZE`? I think that would clarify the code a bit more.

Reply:
Also, maybe you can consider something like `NULLABLE_OFFSET` instead of `INT32_SIZE`.

Reply:
fixed first comment by [this](67d4f34daa3bbdc02f672fdd9998ade7b2c616ae) commit.

I am not changing the second one since there are a lot of such similar places and we either change all or none, and changing all becomes out of scope of this PR.